intel_extension_for_transformers/llm/runtime/graph/scripts/ci/cpp_graph_inference.sh

```diff
@@ -0,0 +1,401 @@
+    esac
+    shift
+done
+
+declare -p cores_list
+declare -p batch_size_list
+declare -p input_list
+declare -p output_list
+declare -p beam_list
+declare -p extra_precision_list
+
+declare -p ppl_dataset_list
+declare -p ppl_nctx_list
+declare -p ppl_fp32_test
+declare -p ppl_mf16_test
+
+function ppl_eval() {
+    local task_name="$1"
+    local n_cores="$2"
+    local model_path="$3"
+    local quantized_weight_path="$4"
+    local memory_dtype_list=('auto')
+    if [[ "$ppl_mf16_test" = true ]]; then
+        memory_dtype_list+=('f16')
+    fi
+
+    echo "=======  PPL Evaluation Start  ======="
+    for memory_dtype in ${memory_dtype_list[@]}; do
+        for ppl_dataset in ${ppl_dataset_list[@]}; do
+            for ppl_nctx in ${ppl_nctx_list[@]}; do
+                local ppl_task_name="$task_name-ppl-$(basename -- "$ppl_dataset")-nctx$ppl_nctx-M$memory_dtype"
+                echo "***** PPL: $ppl_task_name *****"
+                OMP_NUM_THREADS=$(($n_cores * 1)) numactl -m 0 -C 0-$(($n_cores * 1 - 1)) \
+                    python scripts/perplexity.py --model_name "$model_path" --dataset_name "$ppl_dataset" --quantized_weight_path "$quantized_weight_path" --ctx_size $ppl_nctx --n_threads $n_cores --memory_dtype $memory_dtype 2>&1 |
+                    tee "$WORKSPACE/$ppl_task_name.log"
+                mv out/ppl.png "$WORKSPACE/$ppl_task_name.png"
+                mv out/ppl_data.json "$WORKSPACE/$ppl_task_name.json"
+            done
+        done
+    done
+    echo "=======  PPL Evaluation End  ======="
+}
+
+declare -A model_name_map
+model_name_map["llama-2-7b-chat"]="meta-llama/Llama-2-7b-chat-hf"
+model_name_map["gptj-6b"]="EleutherAI/gpt-j-6b"
+model_name_map["gpt-neox-20b"]="EleutherAI/gpt-neox-20b"
+model_name_map["mpt-7b"]="mosaicml/mpt-7b"
+model_name_map["falcon-7b"]="tiiuae/falcon-7b"
+model_name_map["starcoder-3b"]="bigcode/starcoder"
+model_name_map["bloom-7b"]="bigscience/bloom-7b1"
+model_name_map["opt-1.3b"]="facebook/opt-1.3b"
+model_name_map["dolly-v2-3b"]="databricks/dolly-v2-3b"
+model_name_map["chatglm2"]="THUDM/chatglm2-6b"
+model_name_map["chatglm-6b"]="THUDM/chatglm-6b"
+model_name_map["baichuan2-13b"]="baichuan-inc/Baichuan2-13B-Chat"
+model_name_map["baichuan-13b"]="baichuan-inc/Baichuan-13B-Chat"
+model_name_map["mistral-7b"]="mistralai/Mistral-7B-v0.1"
+model_name_map["qwen-7b"]="Qwen/Qwen-7B-Chat"
+
+function main() {
+    conda_env="$1"
+    model="$2"
+    working_dir="$3"       # The "graph" directory with CMakeFile.txt
+    export log_prefix="$4" # The log url prefix
+    compiler_version="$5"
+    PROMPTS_PATH="$script_dir/cpp_graph_prompts.json"
+
+    # init params
+    precision_list=()
+    requirements_file="requirements.txt" # some models need extra constraints
+
+    model_name="${model_name_map["$model"]}"
+
+    if [[ -n "$local_models" ]]; then
+        model_path=$(python -c "import sys, json; print(json.load(sys.stdin).get('$model_name', '$model_name'))" <"$local_models")
```

Comment:
check whether the path exists

Reply:
I think it would be fine to fail fast if the caller gives a non-exist `$local_models` or a `$local_models` with non-exist path. It may be hard to discover the problem if it just fallback to downloading from huggingface.co.

Reply:
if ```local_models``` not exists, the scripts should throw error directly. we don't expect to download models in CI which is quite long.

Reply:
One may want to use this `HF_HOME` for local test (e.g. on machine 6).

> if `local_models` does not exists, the scripts should throw error directly. we don't expect to download models in CI which is quite long.

It would not be time consuming if HF_HOME is configured carefully. I remembered that there was an attempt that use `model_name` directrly. intel-innersource/frameworks.ai.lpot.lpot-validation#534